add parsing for HTTP method and URL

isaacev opened the pull request.

isaacev committed:
add parsing for HTTP method line

isaacev committed:
add parse_url method

- decomposes a URL into hostname, port, and path
- if no port given in URL, default to `80`
- if no path given in URL, default to `/`

isaacev committed:
use parse_method and parse_url to populate HTTPRequest

freiermuth commented:
Just curious, doesn't str.split() split on whitespace by default? I also think explicitly looking for the protocol, destination, etc. with the if 'X' in tup: might work too (I see you're looking at the length of the tuple parsed out)

isaacev commented:
Regarding `str.split()` you're right. Since the spec defines an HTTP method line to end in CRLF I had thought that by specifying a space character in the call to `str.split()` it would avoid splitting between the HTTP version and the trailing CRLF but it seems that even without any parameters being passed to split the string will only break along space characters.
